openwisp_controller/subnet_division/base/models.py

```diff
@@ -0,0 +1,117 @@
+from ipaddress import ip_network
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from swapper import get_model_name
+
+from openwisp_users.mixins import OrgMixin
+from openwisp_utils.base import TimeStampedEditableModel
+
+from .. import settings as app_settings
+
+
+class AbstractSubnetDivisionRule(TimeStampedEditableModel, OrgMixin):
+    type = models.CharField(max_length=200, choices=app_settings.SUBNET_DIVISION_TYPES)
+    # NOTE: Make master_subnet a cached_property since it is utilized often?
```

Comment:
just ensure that when it is accessed, the queryset calls `select_related('master_subnet')` so we avoid generating too many queries